components/base/wheeled/wheeled_base_test.go

```diff
@@ -353,3 +357,252 @@ func TestValidate(t *testing.T) {
 	test.That(t, deps, test.ShouldResemble, []string{"fl-m", "bl-m", "fr-m", "br-m"})
 	test.That(t, err, test.ShouldBeNil)
 }
+
+func TestSpinWithMSMath(t *testing.T) {
+	// tests getTurnState, if we're atTarget, overshot or travelled the minimum amount
+	for _, stops := range []struct {
```

Comment:
in general, I think having test code be less DRY makes it easier to line check which particular case is failing. But if you'd prefer to keep it DRY, I would request as an alternative separating each scenario into it's own instance of `t.Run`, which should make it almost as easy to understand which case is failing. (Putting this comment once here, but I mean for it to apply for all such for-loops in this file)